docs/source/formulations_example.rst

```diff
@@ -0,0 +1,648 @@
+Make sure to set the bounds wide enough to encompass all anticipated use cases of the templates.
+
+.. note:: In the code examples that follow, imports are assumed to be cumulative.
+    Once an import has been demonstrated, we do not include it in subsequent code blocks.
+
+.. code-block:: python
+
+    from citrine.resources.parameter_template import ParameterTemplate
+    from citrine.resources.property_template import PropertyTemplate
+    from gemd.entity.bounds import RealBounds
+
+    # Assume we have already created a project and dataset, named "project" and "dataset."
+    # Though not demonstrated here, it is useful to give templates custom ids so that they can be easily retrieved.
+    blend_time_template = dataset.parameter_templates.register(
+        ParameterTemplate("blend time", bounds=RealBounds(0, 60.0, "s"))
+    )
+    price_template = dataset.property_templates.register(
+        PropertyTemplate("price", bounds=RealBounds(0, 100, "1/kg"))
+    )
+    sucrose_fraction_template = dataset.property_templates.register(
+        PropertyTemplate("sucrose fraction", bounds=RealBounds(0, 1, ""))
+    )
+    tastiness_template = dataset.property_templates.register(
+        PropertyTemplate("tastiness", bounds=RealBounds(0, 10, ""))
+    )
+
+The attribute templates are attached to relevant object templates.
+For example, the process template to represent blending should include ``blend_time_template`` as a ``parameter``.
+
+Although it contains no attributes, we should particular pay attention to the templates that represent the mixing processes.
+These will be used to convert GEMD data into formulations.
+It is possible to use different process templates to distinguish between different types of mixing, but here we use the one template to represesnt both types of mixing that occur (mixing the simple syrup and mixing the margarita ingredients).
+The template includes a comprehensive list of all allowed names and labels.
+
+.. code-block:: python
+
+    from citrine.resources.process_template import ProcessTemplate
+
+    mix_template = dataset.process_templates.register(
+        ProcessTemplate(
+            "mix"
+            allowed_names=["simple syrup", "sugar", "water", "lime juice", "triple sec", "tequila", "ice"],
+            allowed_labels=["sweetener", "acid", "alcohol", "simple syrup"]
+        )
+    )
+
+Notice that "simple syrup" is an allowed ingredient name, but we do not distinguish between "simple syrup A" and "simple syrup B."
+This reflects the fact that we only ever use one simple syrup in a mixing process.
+The resulting table will have one column indicating which simple syrup was used and a second column indicating how much was used.
+Distinguishing in this way between the unique id of the material ("simple syrup A," "simple syrup B," etc.) and the generic name of the ingredient ("simple syrup") is especially useful when there are many materials to choose from.
```

Comment:
MAYBE - I wonder if we should have a break-out box on naming ingredients. This is a common way that people build debt in their GEMD structure. I've been documenting some of the paths forward for DS/DE [here](https://docs.google.com/document/d/1oDlFkXcO570T7yJM1E_xUs32Le1VEC9ssM0o9w7qjA4/edit#heading=h.5a9yl832y301). 

The main points we might want to get across is that:
* The Material UID will correspond to the most specific quantity you can use
* The Ingredient Name will correspond to how ingredients are grouped into columns
* The Ingredient Label(s) will allow grouping of multiple Material UID's when building models and design spaces.